client/tracing/src/block/mod.rs

```diff
@@ -269,7 +269,7 @@ impl<Block, Client> BlockExecutor<Block, Client>
 		tracing::debug!(target: "state_tracing", "Captured {} spans and {} events", spans.len(), events.len());
 
 		let approx_payload_size = BASE_PAYLOAD + events.len() * AVG_EVENT + spans.len() * AVG_SPAN;
-		let response = if approx_payload_size > MAX_PAYLOAD {
+		let response = if approx_payload_size > DEFAULT_MAX_PAYLOAD {
```

Comment:
There's no way to the actual `max payload size` here (as it could be configured by the `CLI parameter)?

/cc @emostov 
I think this would be quite confusing for the users, maybe remove this check if it's not possible to fetch the actual `max_payload_size` configured?

Reply:
Are you saying remove the check all together? Then when are back to the issue with silent errors, right?

I agree this could definitely get confusing, especially if the users set a value < DEFAULT_MAX_PAYLOAD, in which case this would still silently fail. But I would guess 99% of the time this check is still helpful to give a rich error.

Reply:
> Are you saying remove the check all together? Then when are back to the issue with silent errors, right?

yeah, that's was my suggestion but the trade-off is to either reject valid responses that where `value >  DEFAULT_MAX_PAYLOAD` with a nice error message or "silently drop/kill the connection" on responses that exceeds the `actual max limit`.

Not sure what's best but I would be really annoyed that my configured `actual max limit` was not respected and rejected because it was bigger than `DEFAULT_MAX_PAYLOAD`.

I don't know really, so I trust your opinion because you know the sidecar/UX client side better on state tracing.

Reply:
ATM api-sidecar does not tolerate the silent error well and will just hang indefinitely, so I prefer the explicit error + default max payload.

Most optimal solution IMHO would be to import the user configured max payload, but sounds like that is not feasible (as you allude to in the first comment).

Reply:
Ok, I see let's keep it as it is then if can't come up with a better solution.

//cc @kianenigma 

Do you think we could add `max_rpc_payload` field to the [ClientConfig](https://docs.rs/sc-service/0.9.0/sc_service/struct.ClientConfig.html) and expose it from the `Client` via method or something?

I think that would work because `max_rpc_payload` is part of the [Configuration](https://docs.rs/sc-service/0.9.0/sc_service/config/struct.Configuration.html) and it's used in the builder in sc_service?!

Reply:
> Do you think we could add `max_rpc_payload` field to the [ClientConfig](https://docs.rs/sc-service/0.9.0/sc_service/struct.ClientConfig.html) and expose it from the `Client` via a method or something?
> 
> I think that would work because `max_rpc_payload` is part of the [Configuration](https://docs.rs/sc-service/0.9.0/sc_service/config/struct.Configuration.html) and it's used in the builder in sc_service?!

That sounds like a bad idea.

Passing the real `max_rpc_payload` sounds reasonable. The correct solution to this would be to pass the `max_rpc_payload` to `FullState` that provides the `trace_block` rpc method and forward this parameter to the `BlockExecutor` on construction.

Reply:
cool, thanks

Reply:
as far as I can see, the conclusion is to feed the custom value in here. I am ooo for some time, so if anyone wants to fix this in the meantime, they are welcome to take a stab and push directly to the branch. Else I'll take care of it later.

Reply:
@kianenigma I can try and take a stab in the next few days

Reply:
implemented my understanding of @bkchr's suggestion